clang/lib/AST/ExprConstant.cpp

```diff
@@ -15858,7 +15858,7 @@ static bool FastEvaluateAsRValue(const Expr *Exp, Expr::EvalResult &Result,
   }
 
   if (const auto *CE = dyn_cast<ConstantExpr>(Exp)) {
-    if (CE->hasAPValueResult()) {
+    if (CE->hasAPValueResult() && !CE->getAPValueResult().isLValue()) {
```

Comment:
It looks like `ConstantExpr::getAPValueResult()` will sometimes create a new APValue, so we should try ti call `getAPValueResult()` only once.